griptape/utils/decorators.py

```diff
@@ -26,3 +30,49 @@ def wrapper(self, *args, **kwargs):
         return wrapper
 
     return decorator
+
+
+T = TypeVar("T", bound=Callable)
+
+
+def observable(*args: T | Any, **kwargs: Any) -> T:
+    return cast(T, Observable(*args, **kwargs))
+
+
+class Observable:
```

Comment:
I would pull this out into a separate file under `griptape.observability` for consistency.

Reply:
I moved the decorator into a separate file under `griptape.common`. Moving it under `griptape.observability` introduced a particularly difficult-to-mitigate circular dependency. I chose `griptape.common` over `griptape.utils` since it is being introduced in https://github.com/griptape-ai/griptape/pull/861 and seems like a better place for common "core" things.